torch/csrc/utils/python_scalars.h

```diff
@@ -40,7 +40,10 @@ inline PyObject* load_scalar(void* data, at::ScalarType scalarType) {
     case at::kHalf: return PyFloat_FromDouble(at::convert<double, at::Half>(*(at::Half*)data));
     case at::kFloat: return PyFloat_FromDouble(*(float*)data);
     case at::kDouble: return PyFloat_FromDouble(*(double*)data);
-    case at::kComplexFloat: return PyComplex_FromCComplex(*reinterpret_cast<Py_complex *>((std::complex<float>*)data));
+    case at::kComplexFloat: {
+      std::complex<float>* data_ = (std::complex<float>*)data;
+      return PyComplex_FromDoubles(data_->real(), data_->imag());
+    }
     case at::kComplexDouble: return PyComplex_FromCComplex(*reinterpret_cast<Py_complex *>((std::complex<double>*)data));
```

Comment:
@anjali411,

Is this a better way to pass complex data into a python data type? Should you also do the same kComplexDouble?

Reply:
both of them return a  new Python complex number object, just that they take different arguments.

Reply:
The direct struct reader only accepts doubles, otherwise we have to unpack it, alas.